lib/charms/operator_libs_linux/v0/sysctl.py

```diff
@@ -0,0 +1,237 @@
+
+logger = logging.getLogger(__name__)
+
+# The unique Charmhub library identifier, never change it
+LIBID = "17a6cd4d80104d15b10f9c2420ab3266"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 2
+
+
+SYSCTL_DIRECTORY = "/etc/sysctl.d"
+SYSCTL_FILENAME = f"{SYSCTL_DIRECTORY}/95-juju-sysctl.conf"
+SYSCTL_HEADER = f"""# This config file was produced by sysctl lib v{LIBAPI}.{LIBPATCH}
+#
+# This file represents the output of the sysctl lib, which can combine multiple
+# configurations into a single file like this with the help of validation rules. Such
+# rules are used to automatically resolve simple conflicts between two or more charms
+# on one host and is defined as a comment after the configuration option, see example
+# below.
+#
+# Description of validation rules to check if values are valid.
+#   - "range(a,b)" - all numbers between 'a' (included) and 'b' (excluded)
+#   - "a|b|c" - choices 'a', 'b' and 'c'
+#   - "*" - any value
+#   - "" or no comment - only current value, same as "<current value>"
+#   - "disable" - This value will be ignored in any validation and should only be used by
+#                 the charm operator manually or via a charm to override any system
+#                 configuration.
+# Examples:
+# # any value in [10, 50] is valid value
+# vm.swappiness=10  # range(10,51)
+#
+# # 60 and 80 are valid values
+# vm.dirty_ratio = 80  # 60|80
+#
+# # any value is valid
+# vm.dirty_background_ratio = 3  # *
+#
+# # only value 1 is valid value
+# net.ipv6.conf.all.use_tempaddr = 1
+#
+# # validation is disabled
+# net.ipv6.conf.default.use_tempaddr = 0  # disable
+"""
+
+
+class Error(Exception):
+    """Base class of most errors raised by this library."""
+
+    def __repr__(self):
+        """Represent the Error."""
+        return "<{}.{} {}>".format(type(self).__module__, type(self).__name__, self.args)
+
+    @property
+    def name(self):
+        """Return a string representation of the model plus class."""
+        return "<{}.{}>".format(type(self).__module__, type(self).__name__)
+
+    @property
+    def message(self):
+        """Return the message passed as an argument."""
+        return self.args[0]
+
+class SysctlError(Error):
+    """Raised when there's an error running sysctl command."""
+
+
+class SysctlPermissionError(Error):
+    """Raised when there's an error applying values in sysctl."""
+
+
+class ValidationError(Error):
+    """Raised when there's an error in the validation process."""
+
+
+@dataclass()
+class ConfigOption:
+    """Definition of a config option.
+    
+    NOTE: this class can be extended to handle the rule tied to each option.
+    """
+    name: str
+    value: int
+
+    @property
+    def string_format(self) -> str:
+        return f"{self.name}={self.value}"
+
+
+class SysctlConfig:
```

Comment:
I think what we are missing here is to provide current configuration e.g. `SysctlConfig(...).config` or `SysctlConfig(...)[key]`, and function update should return list of changed values or we should be able to access it somehow. What do you think?

Reply:
Changed the class to be a Map. Thanks for the help on this :)